eng/pipelines/templates/scripts/replace-relative-links.yml

```diff
@@ -29,8 +29,8 @@ steps:
             "https://github.com/{repo_id}/tree/{build_sha}/{target_resource_path}"
         )
 
-        LINK_DISCOVERY_REGEX = r"\[([^\]]*)\]\(([^)]*)\)"
-        PREDEFINED_LINK_DISCOVERY_REGEX = r"(\[[^\]]*]\:)\s*([^\s]+)"
+        LINK_DISCOVERY_REGEX = r"\[([^\]]*)\]\(([^)]+)\)"
```

Comment:
I agree with your assessment that this might fix the very specific issue it will definitely not fix the general issue. I'm curious if you have considered using a markdown parsing library instead of a simple regex?

Reply:
I think that's the way we'll have to go here. Context-free will only get us so far.